core/src/main/java/bio/terra/pearl/core/service/search/sql/EnrolleeSearchQueryBuilder.java

```diff
@@ -68,9 +68,19 @@ public SelectQuery<Record> toQuery(DSLContext context, EnrolleeSearchOptions opt
                         condition("enrollee.study_environment_id = ?", studyEnvId)
                 );
 
+
         if (Objects.nonNull(opts.getSortField()) && !opts.getSortField().isEmpty()) {
             String sortField = opts.getSortField() + " " + (opts.isSortAscending() ? "ASC" : "DESC");
-            return selectConditionStep.orderBy(field(sortField)).getQuery();
+
+            var ordered = selectConditionStep.orderBy(field(sortField));
+            if (Objects.nonNull(opts.getLimit())) {
+                return ordered.limit(opts.getLimit()).getQuery();
+            }
+            return ordered.getQuery();
+        }
+
+        if (Objects.nonNull(opts.getLimit())) {
+            return selectConditionStep.limit(opts.getLimit()).getQuery();
         }
```

Comment:
jooq is great, but because java is strict about types and jooq seems to change types every time you call a function on the query, it's really hard to have multiple conditional branches on the same query without a lot of reused code. It's possible I'm missing some easy way to do this